client/app/src/Controller/Report/ReportController.php

```diff
@@ -170,6 +170,58 @@ public function indexAction(Redirector $redirector, ParameterStoreService $param
         ];
     }
 
+    /**
+     * List of reports.
+     *
+     * @Route("/choose-a-client", name="choose_a_client")
+     *
+     * @Security("is_granted('ROLE_LAY_DEPUTY')")     *
+     *
+     * @Template("@App/Index/choose-a-client.html.twig")
+     *
+     * @return array|RedirectResponse
+     */
+    public function chooseAClientAction(Redirector $redirector, ParameterStoreService $parameterStoreService)
+    {
+        $user = $this->userApi->getUserWithData(['user-clients', 'client']);
+        $isMultiClientFeatureEnabled = $parameterStoreService->getFeatureFlag(ParameterStoreService::FLAG_MULTI_ACCOUNTS);
+        file_put_contents('php://stderr', print_r('**** Redirected in chooseAClient **** '.$isMultiClientFeatureEnabled, true));
+
+        if ('1' == $isMultiClientFeatureEnabled) {
+            // redirect back to log out page if signing in with non-primary account with primary email
+            if (!$user->getIsPrimary()) {
+                $primaryEmail = $this->userApi->returnPrimaryEmail($user->getDeputyUid());
+
+                $this->addFlash('nonPrimaryRedirect',
```

Comment:
Do we need to add something to check if there is an actual primary address? What happens should be show if there isn't a primary email?

Reply:
This also returns an array rather than an email by the looks of it.